### PR TITLE
HPCC-20757 Implement new features in Regression Test Engine

### PR DESCRIPTION
**Pull request by AttilaVamos**
Enable multiple instance execution of a same test case (this need for
performance test to execute same test case in multiple times to measure
execution time immediately after clear system disk cache, then again to
measure the impact of caching.

Enable flush disk cache policy definition like:

 -   clear disk cache before the first execution only
 -   clear disk cache before every execution

Signed-off-by: AttilaVamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested it manually

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman please check.

File: testing/regress/README.rst [hunk 1]
```diff
@@ -72,8 +72,15 @@ Result:
 |        --jobnamesuffix suffix   Specify workunit job name suffix.
 |        --flushDiskCache         Flush OS (Linux) Disk Cache before execute ECL code
 |                                 (sudo privileges needed). Ignored when --pq <n> > 1
+|        --flushDiskCachePolicy flushDiskCachePolicy
```
Comment (richardkchapman) [inline testing/regress/README.rst]: alignment

File: testing/regress/README.rst [hunk 2]
```diff
@@ -72,8 +72,15 @@ Result:
...
+|                                 Set flush disk cache policy. The default is 1 to clear
+|                                 before firts run only. For 2 to clear before every
```
Comment (richardkchapman) [inline testing/regress/README.rst]: typo firts should say first

File: testing/regress/README.rst [hunk 3]
```diff
@@ -120,6 +127,12 @@ Result:
 |                                 Specify workunit job name suffix.
 |        --flushDiskCache         Flush OS (Linux) Disk Cache before execute ECL code
 |                                 (sudo privileges needed). Ignored when --pq <n> > 1
+|        --flushDiskCachePolicy flushDiskCachePolicy
```
Comment (richardkchapman) [inline testing/regress/README.rst]: as above

File: testing/regress/README.rst [hunk 4]
```diff
@@ -120,6 +127,12 @@ Result:
...
+|                                 Set flush disk cache policy. The default is 1 to clear
+|                                 before firts run only. For 2 to clear before every
```
Comment (richardkchapman) [inline testing/regress/README.rst]: as above

File: testing/regress/ecl-test [hunk 5]
```diff
@@ -200,6 +200,10 @@ class RegressMain:
                                 metavar="suffix")
         executionParser.add_argument('--flushDiskCache', help="Flush OS (Linux) Disk Cache before execute ECL code (sudo privileges needed). Ignored when --pq <n>  > 1"
                                 , action = 'store_true')
+        executionParser.add_argument('--flushDiskCachePolicy', help="Set flush disk cache policy. The default is 1 to clear before firts run only. For 2 to clear before every runs.",
```
Comment (richardkchapman) [inline testing/regress/ecl-test]: typo firts

**Comment by richardkchapman** [comment]
I didn't see any code that actually did the disk cache flush?

**Comment by AttilaVamos** [comment]
It is already done in HPCC-19267 Add a flushDiskCache option option switchable by //version in to regression suite

File: testing/regress/README.rst (hunk 1 above)
Comment (AttilaVamos) [reply testing/regress/README.rst]: Changed

File: testing/regress/README.rst (hunk 2 above)
Comment (AttilaVamos) [reply testing/regress/README.rst]: Cahnged

File: testing/regress/README.rst (hunk 3 above)
Comment (AttilaVamos) [reply testing/regress/README.rst]: Changed

File: testing/regress/README.rst (hunk 4 above)
Comment (AttilaVamos) [reply testing/regress/README.rst]: Changed

File: testing/regress/ecl-test (hunk 5 above)
Comment (AttilaVamos) [reply testing/regress/ecl-test]: Changed

File: testing/regress/README.rst [hunk 6]
```diff
@@ -70,10 +70,13 @@ Result:
 |        --handleEclccWarningFile, -w
 |                                 Create/overwrite/delete ECLCC warning file.
 |        --jobnamesuffix suffix   Specify workunit job name suffix.
-|        --flushDiskCache         Flush OS (Linux) Disk Cache before execute ECL code
-|                                 (sudo privileges needed). Ignored when --pq <n> > 1
+|        --flushDiskCache         Flush OS (Linux) Disk Cache before execute ECL code (sudo privileges needed). Ignored when --pq <n> > 1
+|        --flushDiskCachePolicy flushDiskCachePolicy
+|                                 Set flush disk cache policy. The default is 1 to clear before first run only. For 2 to clear before every runs.
```
[the remark this answers is not in the document]
Comment (AttilaVamos) [reply testing/regress/README.rst]: Typo is fixed, but I can't see the alignment problem.

**Comment by AttilaVamos** [comment]
@richardkchapman please hold it a bit more.

**Comment by AttilaVamos** [comment]
@richardkchapman please check again.

**Pull request merged**